src/liveshare/shareCommands.ts

```diff
@@ -0,0 +1,161 @@
+        // to the guest session.
+        [Callback.GetFileContent]: async (args: [text: string, encoding?: string]): Promise<string | Buffer> => {
+            return args[1] !== undefined ?
+                await fs.readFile(args[0], args[1]) :
+                await fs.readFile(args[0]);
+        }
+    },
+    'guest': {
+        [Callback.NotifyRequestUpdate]: (args: [file: string, force: boolean]): void => {
+            void updateGuestRequest(args[0], args[1]);
+        },
+        [Callback.NotifyEnvUpdate]: (args: [hostEnv: string]): void => {
+            void updateGuestGlobalenv(args[0]);
+        },
+        [Callback.NotifyPlotUpdate]: (args: [file: string]): void => {
+            void updateGuestPlot(args[0]);
+        },
+        [Callback.NotifyGuestPlotManager]: (args: [url: string]): void => {
+            globalHttpgdManager?.showViewer(args[0]);
+        },
+        [Callback.OrderDetach]: (): void => {
+            void detachGuest();
+        },
+        /// vscode Messages ///
+        // The host sends messages to the guest, which are displayed as a vscode window message
+        // E.g., teling the guest a terminal is not attached to the current session
+        // This way, we don't have to do much error checking on the guests side, which is more secure
+        // and less prone to error
+        [Callback.NotifyMessage]: (args: [text: string, messageType: MessageType]): void => {
+            switch (args[1]) {
+                case MessageType.error:
+                    return void vscode.window.showErrorMessage(args[0]);
+                case MessageType.information:
+                    return void vscode.window.showInformationMessage(args[0]);
+                case MessageType.warning:
+                    return void vscode.window.showWarningMessage(args[0]);
+                case undefined:
+                    return void vscode.window.showInformationMessage(args[0]);
+            }
+        }
+    }
+};
+
+
+// The following onRequest and request methods are wrappers
+// around the vsls RPC API. These are intended to simplify
+// the API, so that the learning curve is minimal for contributing
+// future callbacks.
+//
+// You can see that the onNotify and notify methods have been
+// aggregated under these two methods. This is because the host service
+// has no request methods, and for *most* purposes, there is little functional
+// difference between request and notify.
+export function onRequest(name: string, command: unknown, service: vsls.SharedService | vsls.SharedServiceProxy | null): void {
+    if (isGuest()) {
+        // is guest service
+        (service as vsls.SharedServiceProxy).onNotify(name, command as vsls.NotifyHandler);
+    } else {
+        // is host service
+        (service as vsls.SharedService).onRequest(name, command as vsls.RequestHandler);
+    }
+}
+
+export function request(name: string, ...rest: unknown[]): unknown {
```

Comment:
To avoid conusion, should we maybe use a more descriptive name? E.g. `liveShareRequest` etc.